File: 4000-largest-integer-with-given-digit-sum/4000-largest-integer-with-given-digit-sum.cpp
class Solution {
public:
    int largestInteger(int n, int s) {
        if(s > 9*n)
        {
            return -1;
        }
        int lar=0;
        for(int i=0; i<n; i++)
        {
            int d = (s>=9) ? 9:s;
            lar = lar*10 + d;
            s-=d;
        }
        return lar;
    }
};

File: 4000-largest-integer-with-given-digit-sum/4000-largest-integer-with-given-digit-sum.py
class Solution:
    def largestInteger(self, n: int, s: int) -> int:
        if(s > 9*n):
        
            return -1
        
        lar=0
        for i in range(0, n):
        
            d = 9 if (s>=9) else s
            lar = lar*10 + d
            s-=d
        
        return lar